### PR TITLE
UI-Upgrade in Administration/Practice

### PR DESCRIPTION
**Pull request by nileshprasad137**
Bootstrap included in `Administration/Practice`.

### CODE REVIEW

File: templates/documents/general_queue.php [hunk 1]
```diff
@@ -12,19 +12,19 @@ function submit_documents()
 
 <a href="controller.php?practice_settings&<?php echo $this->top_action; ?>document_category&action=list" 
    onclick="top.restoreSession()" class="css_button" >
-<span>Edit Categories</span></a><a href="#" onclick="submit_documents();" class="css_button" target="_self" ><span>Update files</span></a>
-<input type="hidden" name="process" value="<?php echo self::PROCESS;?>" /><br><br>
+<span><?php echo xl("Edit Categories");?></span></a><a href="#" onclick="submit_documents();" class="css_button" target="_self" ><span>Update files</span></a>
```
Comment (teryhill) [inline templates/documents/general_queue.php]: I think these should be xlt instead of xl

File: templates/documents/general_queue.php [hunk 2]
```diff
@@ -12,19 +12,19 @@ function submit_documents()
...
-    <tr class="showborder_head">
-        <th colspan="2" width="110px">Name</td>
-        <th width="100px">Date</td>
-        <th width="200px">Patient</td>
-        <th colspan="2">Category</td>
+    <tr>
+        <th colspan="2"><?php echo xl("Name");?></td>
+        <th><?php echo xl("Date");?></td>
+        <th><?php echo xl("Patient");?></td>
+        <th colspan="2"><?php echo xl("Category");?></td>
```
Comment (teryhill) [inline templates/documents/general_queue.php]: Same here

File: templates/documents/general_queue.php [hunk 3]
```diff
@@ -58,7 +58,7 @@ function submit_documents()
    <?php }    
    } else{ ?>
         <tr height="25" class="center_display">
-                <td colspan="6"><?php echo "No Documents Found";?></td>
+                <td colspan="6"><?php echo xl("No Documents Found");?></td>
```
Comment (teryhill) [inline templates/documents/general_queue.php]: Here and line 37 also

File: templates/hl7/general_parse.php [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+<table class="table">
 <!--<tr><td>Example HL7 data<td></tr>
 <tr><td>MSH|^~\&|ADT1|CUH|LABADT|CUH|198808181127|SECURITY|ADT^A01|MSG00001|P|2.3|
 EVN|A01|198808181122||
 PID|||PATID1234^5^M11||RYAN^HENRY^P||19610615|M||C|1200 N ELM STREET^^GREENSBORO^NC^27401-1020|GL|(919)379-1212|(919)271-3434 ||S||PATID12345001^2^M10|123456789|987654^NC|
 NK1|JOHNSON^JOAN^K|WIFE||||||NK^NEXT OF KIN
 PV1|1|I|2000^2053^01||||004777^FISHER^BEN^J.|||SUR||||ADM|A0|</td></tr>-->
 <tr>
-<td colspan="2"><b>Paste HL7 Data</b></td>
+<td colspan="2"><b><?php echo xl("Paste HL7 Data");?></b></td>
```
Comment (teryhill) [inline templates/hl7/general_parse.php]: xlt

File: templates/insurance_companies/general_edit.php [hunk 5]
```diff
@@ -24,70 +24,70 @@ function jsWaitForDelay(delay) {
...
+    <td>
+        <input type="text" class="form-control" size="40" name="attn" value="<?php echo $this->insurancecompany->get_attn();?>" onKeyDown="PreventIt(event)" />
     </td>
 </tr>
 
 <tr>
-    <td VALIGN="MIDDLE" ><?php echo xl("Address");?> </td>
-    <td VALIGN="MIDDLE" >
-        <input type="text" size="40" name="address_line1" value="<?php $this->insurancecompany->address->line1;?>" 
+    <td><?php echo xl("Address(line1)");?> </td>
```
Comment (teryhill) [inline templates/insurance_companies/general_edit.php]: the translation engine will have trouble with the punctuation in side of the statement Maybe something like 
xlt("Address". ( . "line1". ) .") Not real sure about the syntax.

File: templates/insurance_companies/general_edit.php [hunk 6]
```diff
@@ -24,70 +24,70 @@ function jsWaitForDelay(delay) {
...
+    <td>
+        <input type="text" class="form-control" size="40" name="address_line1" value="<?php $this->insurancecompany->address->line1;?>" 
                onKeyDown="PreventIt(event)" />
     </td>
 </tr>
 <tr>
-    <td VALIGN="MIDDLE" ><?php echo xl("Address");?> </td>
-    <td VALIGN="MIDDLE" >
-        <input type="text" size="40" name="address_line2" value="<?php echo $this->insurancecompany->address->line2;?>" 
+    <td><?php echo xl("Address(line2)");?> </td>
```
Comment (teryhill) [inline templates/insurance_companies/general_edit.php]: Same here

File: templates/insurance_companies/general_edit.php [hunk 7]
```diff
@@ -102,16 +102,16 @@ function jsWaitForDelay(delay) {
...
-        <input type="text" name="x12_receiver_id" value="{$insurancecompany->get_x12_receiver_id()}" >
+    <td COLSPAN="1" ALIGN="LEFT" >X12 Receiver ID</td>
+    <td COLSPAN="2" ALIGN="LEFT" >
+        <input type="text" class="form-control" name="x12_receiver_id" value="{$insurancecompany->get_x12_receiver_id()}" >
     </td>
 </tr>-->
 <tr>
-    <td VALIGN="MIDDLE">Default X12 Partner</td>
-    <td VALIGN="MIDDLE">
+    <td><?php echo xl("Default X12 Partner");?></td>
```
Comment (teryhill) [inline templates/insurance_companies/general_edit.php]: xlt

File: templates/insurance_companies/general_edit.php [hunk 8]
```diff
@@ -125,9 +125,9 @@ function jsWaitForDelay(delay) {
 </tr>
 <tr height="25"><td colspan=2>&nbsp;</td></tr>
 <tr>
-    <td colspan="2"><a href="javascript:submit_insurancecompany();" class="css_button"><span>Save</span></a>
+    <td colspan="2"><a href="javascript:submit_insurancecompany();" class="css_button"><span><?php echo xl("Save");?></span></a>
         <a href="controller.php?practice_settings&insurance_company&action=list" class="css_button" onclick="top.restoreSession()">
-            <span>Cancel</span></a></td>
+            <span><?php echo xl("Cancel");?></span></a></td>
```
Comment (teryhill) [inline templates/insurance_companies/general_edit.php]: xlt here and line 128

File: templates/insurance_companies/general_list.php [hunk 9]
```diff
@@ -2,28 +2,28 @@
...
 <br>
-<table cellpadding="1" cellspacing="0" class="showborder">
-    <tr class="showborder_head">
-        <th width="140px"><b><?php echo xl("Name");?> </b></th>
-        <th width="300px"><b><?php echo xl("City,State");?> </b></th>
+<table class="table table-hover ">
+    <tr>
+        <th><b><?php echo xl("Name");?> </b></th>
+        <th><b><?php echo xl("City,State");?> </b></th>
         <th><b><?php echo xl("Default X12 Partner");?> </b></th>
```
Comment (teryhill) [inline templates/insurance_companies/general_list.php]: xlt and punctuation

File: templates/insurance_numbers/general_edit.php [hunk 10]
```diff
@@ -10,24 +10,24 @@
...
-<tr  class="showborder_head">
-    <th class="small">Company Name</th>
-    <th class="small">Provider Number</th>
-    <th class="small">Rendering Provider Number</th>
-    <th class="small">Group Number</th>
+<tr>
+    <th><?php echo xl("Company Name");?></th>
+    <th><?php echo xl("Provider Number");?></th>
+    <th><?php echo xl("Rendering Provider Number");?></th>
+    <th><?php echo xl("Group Number");?></th>
```
Comment (teryhill) [inline templates/insurance_numbers/general_edit.php]: xlt

File: templates/insurance_numbers/general_edit.php [hunk 11]
```diff
@@ -39,43 +39,35 @@
 <?php } } 
 else { ?>
 <tr>
-   <td colspan="5">No entries found, use the form below to add an entry.</td>
+   <td colspan="5"><?php echo xl("No entries found, use the form below to add an entry.");?></td>
```
Comment (teryhill) [inline templates/insurance_numbers/general_edit.php]: xlt

File: templates/insurance_numbers/general_edit.php [hunk 12]
```diff
@@ -39,43 +39,35 @@
...
-<?php echo $_GET['showform'];?>
 <?php if($_GET['showform'] == 'true') 
 {
 ?>
 <tr> <td style="border-style:none;" colspan="5">
 <br>
 <b><span>
         <?php if ($this->ins->get_id() == "") { echo "Add Provider Number"; }
-            else { echo "Update Provider Number"; } 
+            else { echo xl("Update Provider Number"); }
```
Comment (teryhill) [inline templates/insurance_numbers/general_edit.php]: xlt and 51

File: templates/insurance_numbers/general_edit.php [hunk 13]
```diff
@@ -118,20 +110,20 @@ class="css_button_small" style='margin-top:2px'
         </td>
 </tr>
 <tr>
-        <td style="border-style:none;" VALIGN="MIDDLE" >Rendering Provider Number</td>
-        <td style="border-style:none;" VALIGN="MIDDLE" >
-                <input type="text" size="20" name="rendering_provider_number" value="<?php echo $this->ins->get_rendering_provider_number();?>" 
+        <td><?php echo xl("Rendering Provider Number");?></td>
```
Comment (teryhill) [inline templates/insurance_numbers/general_edit.php]: xlt lines 86,92

File: templates/insurance_numbers/general_edit.php [hunk 14]
```diff
@@ -118,20 +110,20 @@ class="css_button_small" style='margin-top:2px'
...
+        <td>
+                <input type="text" class="form-control" size="20" name="rendering_provider_number" value="<?php echo $this->ins->get_rendering_provider_number();?>" 
                        onKeyDown="PreventIt(event)" />
         </td>
 </tr>
 <tr>
-        <td style="border-style:none;" VALIGN="MIDDLE" >Rendering Provider Number Type</td>
-        <td style="border-style:none;" VALIGN="MIDDLE" >
+        <td><?php echo xl("Rendering Provider Number Type");?></td>
```
Comment (teryhill) [inline templates/insurance_numbers/general_edit.php]: here

File: templates/insurance_numbers/general_edit.php [hunk 15]
```diff
@@ -145,15 +137,15 @@ class="css_button_small" style='margin-top:2px'
         </td>
 </tr>
 <tr>
-        <td style="border-style:none;" VALIGN="MIDDLE" >Group Number</td>
-        <td style="border-style:none;" VALIGN="MIDDLE" >
-                <input type="text" size="20" name="group_number" value="<?php echo $this->ins->get_group_number(); ?>" 
+        <td><?php echo xl("Group Number");?></td>
```
Comment (teryhill) [inline templates/insurance_numbers/general_edit.php]: xlt

File: templates/insurance_numbers/general_list.php [hunk 16]
```diff
@@ -1,15 +1,16 @@
...
-                <th width="100px"><?php echo xl("Rendering");?>  #</th>
-                <th width="100px"><?php echo xl("Group");?>  #</th>
+<div class="table-responsive">
+<table class="table table-hover">
+        <tr>
+                <th><?php echo xl("Name");?></th>
+                <th>&nbsp;</th>
+                <th><?php echo xl("Provider");?>#</th>
+                <th><?php echo xl("Rendering");?>#</th>
+                <th><?php echo xl("Group");?>#</th>
```
Comment (teryhill) [inline templates/insurance_numbers/general_list.php]: xlt

File: templates/pharmacies/general_edit.php [hunk 17]
```diff
@@ -1,58 +1,58 @@
...
+    <td><?php echo xl("Name");?> </td>
+    <td>
+        <input type="text" class="form-control" size="40" name="name" value="<?php echo $this->pharmacy->name;?>" onKeyDown="PreventIt(event)" />(Required)
     </td>
 </tr>
 <tr>
-    <td VALIGN="MIDDLE" ><?php echo xl("Address");?></td>
-    <td VALIGN="MIDDLE" >
-        <input type="text" size="40" name="address_line1" value="<?php echo $this->pharmacy->address->line1;?>" onKeyDown="PreventIt(event)" />
+    <td><?php echo xl("Address(line1)");?></td>
```
Comment (teryhill) [inline templates/pharmacies/general_edit.php]: translation xlt and ()

File: templates/pharmacies/general_edit.php [hunk 18]
```diff
@@ -1,58 +1,58 @@
...
+    <td>
+        <input type="text" class="form-control" size="40" name="address_line1" value="<?php echo $this->pharmacy->address->line1;?>" onKeyDown="PreventIt(event)" />
     </td>
 </tr>
 <tr>
-    <td VALIGN="MIDDLE" ><?php echo xl("Address");?></td>
-    <td VALIGN="MIDDLE" >
-        <input type="text" size="40" name="address_line2" value="<?php echo $this->pharmacy->address->line2;?>" onKeyDown="PreventIt(event)" />
+    <td><?php echo xl("Address(line2)");?></td>
```
Comment (teryhill) [inline templates/pharmacies/general_edit.php]: same as before

File: templates/pharmacies/general_list.php [hunk 19]
```diff
@@ -1,17 +1,18 @@
...
 
-<table cellpadding="1" cellspacing='0' class="showborder">
-    <tr class="showborder_head">
-        <th width="170px"><b><?php echo xl("Name");?></b></th>
-        <th width="320px"><b><?php echo xl("Address");?></b></th>
+<div class="table-responive">
+<table class="table table-hover">
+    <tr>
+        <th><b><?php echo xl("Name");?></b></th>
+        <th><b><?php echo xl("Address");?></b></th>
```
Comment (teryhill) [inline templates/pharmacies/general_list.php]: translation

File: templates/x12_partners/general_edit.php [hunk 20]
```diff
@@ -175,24 +175,24 @@ function Waittoredirect(delaymsec) {
...
 
                 <a href="controller.php?practice_settings&x12_partner&action=list"  class="css_button" onclick="top.restoreSession()">
-                    <span>Cancel</span></a></td>
+                    <span><?php echo xl("Cancel");?></span></a></td>
         </tr>
     </table>
     <input type="hidden" name="id" value="<?php echo $this->partner->id; ?>" />
     <input type="hidden" name="process" value="<?php echo self::PROCESS; ?>" />
     <input type="hidden" name="sub" value="no" />
-</form>
```
Comment (teryhill) [inline templates/x12_partners/general_edit.php]: check translations

File: templates/x12_partners/general_list.php [hunk 21]
```diff
@@ -1,10 +1,10 @@
...
-        <th width="200px"><?php echo xl("Name");?></th>
-        <th width="130px"><?php echo xl("Sender ID");?></th>
-        <th width="130px"><?php echo xl("Receiver ID");?></th>
+<span><?php echo xl("Add New Partner");?></span></a><br><br>
+<table class="table table-hover">
+    <tr>
+        <th><?php echo xl("Name");?></th>
+        <th><?php echo xl("Sender ID");?></th>
+        <th><?php echo xl("Receiver ID");?></th>
         <th><?php echo xl("Version");?></th>
```
Comment (teryhill) [inline templates/x12_partners/general_list.php]: translations

File: templates/documents/general_queue.php (hunk 1 above)
Comment (nileshprasad137) [inline templates/documents/general_queue.php]: Alright! I had seen `library/translation.inc.php` , so thought of using `xl(string)`. Can you elaborate on some differences between xl and xlt or maybe a link to docs would also be enough.

File: templates/x12_partners/general_list.php (hunk 21 above)
Comment (aethelwulffe) [inline templates/x12_partners/general_list.php]: Yes, xlt, not xl

Comment (nileshprasad137) [inline templates/x12_partners/general_list.php]: Alright. Doing it.

**Comment by nileshprasad137** [comment]
I checked for "Hindi" , works well for most cases.

**Comment by nileshprasad137** [comment]
However, I noticed one thing, when I log in with a different language, it shows the proper translation but when I reload the page, translation vanishes and English is shown again.

**Comment by aethelwulffe** [comment]
Sounds more like a browser issue.  XAMPP + Firefox, with a "wipe history when closed" setup works just fine.

**Comment by aethelwulffe** [comment]
Slightly off-topic question:
Nilesh, how many languages of the sub-continent could you get translation help with from medically trained folks (such as at your university)?  To me, it is scary that we are constantly putting xlt() on stuff that we don't actually have translations for.  Hindi, Urdu, Tamil...all very important with open-source tools.  How important do you think Devanagari script would be for the acceptance and use of the application?

**Comment by nileshprasad137** [comment]
@aethelwulffe I have very less knowledge over how translations work. I am just going to enter my 3rd yr. of Comp.Sc. program and maybe I can be of help on this topic later on. 
Anyways,

> Nilesh, how many languages of the sub-continent could you get translation help with from medically trained folks (such as at your university)?

Hindi is my mother-tongue, I know Marathi a little, but my friends know that language well. Maybe some Gujarati as well. 
But again, I can't say if I can be on help on this at present date. I would like to know how translations work .

**Comment by aethelwulffe** [comment]
It will take a bit of my own investigation to answer how *we* are going 
to handle it, but here 
<http://www.open-emr.org/wiki/index.php/OpenEMR_Internationalization_Translator_Guide> 
is a link to how it is supposed to currently work:

**Comment by aethelwulffe** [comment]
Unfortunately, my whole two hours of happy quiet computer time for the 
day are officially over.  I should have enough freedom to do some real 
work tomorrow instead of just trying to keep up with communications.  I 
am pretty close to actual frustration at the moment with all the curve 
balls being thrown at me.  To use a baseball expression, despite 
swinging all week, I have yet to hit one pitch, but people are keeping 
me at the plate despite my shitty batting average.  Must have something 
to do with there being a dozen different pitchers all throwing at the 
same time...

**Pull request merged**